tests/unit/drivers/test_encodingevaluation_driver.py

```diff
@@ -0,0 +1,103 @@
+import numpy as np
+from jina.drivers.evaluate import EncodeEvaluationDriver
+from jina.drivers.helper import DocGroundtruthPair, array2pb
+from jina.executors.evaluators.encode import BaseEncodingEvaluator
+from jina.proto import jina_pb2
+
+
+class MockDiffEvaluator(BaseEncodingEvaluator):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    @property
+    def name(self):
+        return 'MockDiffEvaluator'
+
+    def evaluate(self, doc_embedding: 'np.array', groundtruth_embedding: 'np.array', *args, **kwargs) -> float:
+        """"
+        :param doc_embedding: the embedding of the document (resulting from an Encoder)
+        :param groundtruth_embedding: the expected embedding of the document
+        :return the evaluation metric value for the request document
+        """
+        return abs(sum(doc_embedding - groundtruth_embedding) / len(doc_embedding))
```

Comment:
this can be a fixture, because I see this class in three places.

Reply:
@JoanFM ok, because personally I think test should only consist code related to test, you should decoupke anything related to resources, instance construction outside the test itself. Fixture has two fold of benefits 1. reuse 2. more importantly, reflects the objective of **a function should do one thing, and only one thing. ** (quoted from clean code)

Reply:
Ok, I see the point, makes a lot of sense. Thanks for the clarification !!! Good learning for me as well! Will change this!